frontend/src/app/home/Connected.tsx

```diff
@@ -10,41 +10,71 @@ export const provider = new Provider(Network.TESTNET);
 
 export function Connected() {
   const [pet, setPet] = useState<Pet>();
+  const [collectionID, setCollectionID] = useState<string>();
   const { account, network } = useWallet();
 
   const fetchPet = useCallback(async () => {
     if (!account?.address) return;
 
-    const payload = {
+    const getAptogotchiPayload = {
       function: `${process.env.NEXT_PUBLIC_CONTRACT_ADDRESS}::main::get_aptogotchi`,
       type_arguments: [],
       arguments: [account.address],
     };
 
-    const response = await provider.view(payload);
+    const aptogotchiResponse = await provider.view(getAptogotchiPayload);
+
     const noPet = ["", "0", "0", "0x"];
 
-    if (JSON.stringify(response) !== JSON.stringify(noPet)) {
+    if (JSON.stringify(aptogotchiResponse) !== JSON.stringify(noPet)) {
       setPet({
-        name: response[0] as unknown as string,
-        energy_points: parseInt(response[2] as unknown as string),
-        parts: (response[3] as unknown as string)
+        name: aptogotchiResponse[0] as unknown as string,
+        energy_points: parseInt(aptogotchiResponse[2] as unknown as string),
+        parts: (aptogotchiResponse[3] as unknown as string)
           .split("0")
           .slice(2)
           .map(Number),
       });
     }
   }, [account?.address]);
 
+  const fetchCollectionID = useCallback(async () => {
+    if (!account?.address) return;
+
+    const getAptogotchiCollectionIDPayload = {
+      function: `${process.env.NEXT_PUBLIC_CONTRACT_ADDRESS}::main::get_aptogotchi_collection_id`,
+      type_arguments: [],
+      arguments: [],
+    };
+
+    const aptogotchiCollectionIDResponse = (await provider.view(
+      getAptogotchiCollectionIDPayload
+    )) as string[];
+
+    setCollectionID(aptogotchiCollectionIDResponse[0]);
+  }, [account?.address]);
+
   useEffect(() => {
     if (!account?.address || !network) return;
 
     fetchPet();
   }, [account?.address, fetchPet, network]);
 
+  useEffect(() => {
+    if (!account?.address || !network) return;
+
+    fetchCollectionID();
+  }, [fetchCollectionID, network]);
+
   return (
     <div className="flex flex-col gap-3 p-3">
-      {pet ? <Pet pet={pet} setPet={setPet} /> : <Mint fetchPet={fetchPet} />}
+      {collectionID ? (
+        pet ? (
+          <Pet pet={pet} setPet={setPet} collectionID={collectionID} />
```

Comment:
collectionID is not used in `Connected` or `Pet`. Let's just query collectionID in `Collection`.

Reply:
done